activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/bpmn/helper/ErrorPropagation.java

```diff
@@ -276,63 +275,77 @@ protected static Map<String, List<Event>> findCatchingEventsForProcess(
         String processDefinitionId,
         String errorRef
     ) {
-        Map<String, List<Event>> eventMap = new HashMap<String, List<Event>>();
+        Map<String, List<Event>> eventMap = new HashMap<>();
         Process process = ProcessDefinitionUtil.getProcess(processDefinitionId);
         BpmnModel bpmnModel = ProcessDefinitionUtil.getBpmnModel(processDefinitionId);
 
         String compareErrorCode = retrieveErrorCode(bpmnModel, errorRef);
 
+        findCatchingEventSubprocesses(eventMap, process, bpmnModel, compareErrorCode);
```

Comment:
Instead of passing the map as parameter, prefer to instantiate it inside the the method and use it as return type. You can then aggregate the new entries to the exiting on by using `eventMap.putAll` method